src/main/scala/is/hail/expr/Type.scala

```diff
@@ -440,39 +440,47 @@ case class TSet(elementType: Type) extends TIterable {
   override def genValue: Gen[Annotation] = Gen.buildableOf[Set, Annotation](elementType.genValue)
 }
 
-case class TDict(elementType: Type) extends TContainer {
-  override def children = Seq(elementType)
+case class TMap(keyType: Type, valueType: Type) extends TContainer {
 
-  override def unify(concrete: Type) = {
+  def elementType: Type = valueType
+
+  override def children = Seq(keyType, valueType)
+
+  override def unify(concrete: Type): Boolean = {
     concrete match {
-      case TDict(celementType) => elementType.unify(celementType)
+      case TMap(kt, vt) => keyType.unify(kt) && valueType.unify(vt)
       case _ => false
     }
   }
 
-  override def subst() = TDict(elementType.subst())
+  override def subst() = TMap(keyType.subst(), valueType.subst())
 
-  override def toString = s"Dict[$elementType]"
+  override def toString = s"Map[$keyType, $valueType]"
 
   override def pretty(sb: StringBuilder, indent: Int, printAttrs: Boolean, compact: Boolean = false) {
-    sb.append("Dict[")
-    elementType.pretty(sb, indent, printAttrs, compact)
+    sb.append("Map[")
+    keyType.pretty(sb, indent, printAttrs, compact)
+    if (compact)
+      sb += ','
+    else
+      sb.append(", ")
```

Comment:
Does the one space really matter ? :) 
Also, any reason for using `+=` in `if` and `append` in `else` ?